kv/kvserver/protectedts/ptstorage: TestCorruptData failed

kv/kvserver/protectedts/ptstorage.TestCorruptData [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3215587&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3215587&tab=artifacts#/) on master @ [2ca4acb22aade1175232187bb2b1754057e89f9b](https://github.com/cockroachdb/cockroach/commits/2ca4acb22aade1175232187bb2b1754057e89f9b):


```
=== RUN   TestCorruptData
--- FAIL: TestCorruptData (0.33s)
=== RUN   TestCorruptData/corrupt_hlc_timestamp
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCorruptData_corrupt_hlc_timestamp743071762
    test_log_scope.go:74: use -show-logs to present logs inline
E210723 06:26:02.392390 18466 kv/kvserver/protectedts/ptstorage/storage.go:224  [-] 1  failed to parse row as record: failed to parse timestamp for a6a729ae-6379-43ab-81c8-4bd376d46a02: logical part has too many digits
E210723 06:26:02.393380 17261 kv/kvserver/protectedts/ptstorage/storage.go:224  [-] 2  failed to parse row as record: failed to parse timestamp for a6a729ae-6379-43ab-81c8-4bd376d46a02: logical part has too many digits
    storage_test.go:579: 
        	Error Trace:	storage_test.go:579
        	Error:      	"[{ERROR %!s(int64=1627021562393380000) %!s(int64=17261) kv/kvserver/protectedts/ptstorage/storage.go %!s(int64=224) failed to parse row as record: failed to parse timestamp for a6a729ae-6379-43ab-81c8-4bd376d46a02: logical part has too many digits - %!s(uint64=58) %!s(bool=false) DEV %!s(uint32=0) %!s(uint32=0) %!s(uint32=0)} {ERROR %!s(int64=1627021562392390000) %!s(int64=18466) kv/kvserver/protectedts/ptstorage/storage.go %!s(int64=224) failed to parse row as record: failed to parse timestamp for a6a729ae-6379-43ab-81c8-4bd376d46a02: logical part has too many digits - %!s(uint64=57) %!s(bool=false) DEV %!s(uint32=0) %!s(uint32=0) %!s(uint32=0)}]" should have 1 item(s), but has 2
        	Test:       	TestCorruptData/corrupt_hlc_timestamp
    panic.go:613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCorruptData_corrupt_hlc_timestamp743071762
    --- FAIL: TestCorruptData/corrupt_hlc_timestamp (0.16s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestCorruptData PKG=./pkg/kv/kvserver/protectedts/ptstorage TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</p>
</details>
/cc @cockroachdb/kv ajwerner
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCorruptData.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/protectedts/ptstorage.TestCorruptData [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3247421&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3247421&tab=artifacts#/) on master @ [701b177d8f4b81d8654dfb4090a2cd3cf82e63a7](https://github.com/cockroachdb/cockroach/commits/701b177d8f4b81d8654dfb4090a2cd3cf82e63a7):


```
=== RUN   TestCorruptData
--- FAIL: TestCorruptData (0.38s)
=== RUN   TestCorruptData/corrupt_spans
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCorruptData_corrupt_spans188771539
    test_log_scope.go:74: use -show-logs to present logs inline
E210731 06:40:01.216074 17034 kv/kvserver/protectedts/ptstorage/storage.go:224  [-] 1  failed to parse row as record: failed to unmarshal spans for 4bfbed8b-5e7a-4068-9243-c39fe0940fe2: unexpected EOF
E210731 06:40:01.216606 15941 kv/kvserver/protectedts/ptstorage/storage.go:224  [-] 2  failed to parse row as record: failed to unmarshal spans for 4bfbed8b-5e7a-4068-9243-c39fe0940fe2: unexpected EOF
    storage_test.go:527: 
        	Error Trace:	storage_test.go:527
        	Error:      	"[{ERROR %!s(int64=1627713601216606000) %!s(int64=15941) kv/kvserver/protectedts/ptstorage/storage.go %!s(int64=224) failed to parse row as record: failed to unmarshal spans for 4bfbed8b-5e7a-4068-9243-c39fe0940fe2: unexpected EOF - %!s(uint64=58) %!s(bool=false) DEV %!s(uint32=0) %!s(uint32=0) %!s(uint32=0)} {ERROR %!s(int64=1627713601216074000) %!s(int64=17034) kv/kvserver/protectedts/ptstorage/storage.go %!s(int64=224) failed to parse row as record: failed to unmarshal spans for 4bfbed8b-5e7a-4068-9243-c39fe0940fe2: unexpected EOF - %!s(uint64=57) %!s(bool=false) DEV %!s(uint32=0) %!s(uint32=0) %!s(uint32=0)}]" should have 1 item(s), but has 2
        	Test:       	TestCorruptData/corrupt_spans
    panic.go:613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCorruptData_corrupt_spans188771539
    --- FAIL: TestCorruptData/corrupt_spans (0.22s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestCorruptData PKG=./pkg/kv/kvserver/protectedts/ptstorage TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/kv ajwerner
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCorruptData.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/protectedts/ptstorage.TestCorruptData [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3330312&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3330312&tab=artifacts#/) on master @ [7897f24246bef3cb94f9f4bfaed474ecaa9fdee6](https://github.com/cockroachdb/cockroach/commits/7897f24246bef3cb94f9f4bfaed474ecaa9fdee6):


```
=== RUN   TestCorruptData
--- FAIL: TestCorruptData (0.46s)
=== RUN   TestCorruptData/corrupt_spans
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCorruptData_corrupt_spans653245200
    test_log_scope.go:74: use -show-logs to present logs inline
E210820 06:21:49.454581 19189 kv/kvserver/protectedts/ptstorage/storage.go:240  [-] 1  failed to parse row as record: failed to unmarshal spans for 8dd7559c-dccc-49e7-bc86-4e119d3a4ee2: unexpected EOF
E210820 06:21:49.455517 18044 kv/kvserver/protectedts/ptstorage/storage.go:240  [-] 2  failed to parse row as record: failed to unmarshal spans for 8dd7559c-dccc-49e7-bc86-4e119d3a4ee2: unexpected EOF
    storage_test.go:574: 
        	Error Trace:	storage_test.go:574
        	Error:      	"[{ERROR %!s(int64=1629440509455517000) %!s(int64=18044) kv/kvserver/protectedts/ptstorage/storage.go %!s(int64=240) failed to parse row as record: failed to unmarshal spans for 8dd7559c-dccc-49e7-bc86-4e119d3a4ee2: unexpected EOF - %!s(uint64=60) %!s(bool=false) DEV %!s(uint32=0) %!s(uint32=0) %!s(uint32=0)} {ERROR %!s(int64=1629440509454581000) %!s(int64=19189) kv/kvserver/protectedts/ptstorage/storage.go %!s(int64=240) failed to parse row as record: failed to unmarshal spans for 8dd7559c-dccc-49e7-bc86-4e119d3a4ee2: unexpected EOF - %!s(uint64=59) %!s(bool=false) DEV %!s(uint32=0) %!s(uint32=0) %!s(uint32=0)}]" should have 1 item(s), but has 2
        	Test:       	TestCorruptData/corrupt_spans
    panic.go:613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCorruptData_corrupt_spans653245200
    --- FAIL: TestCorruptData/corrupt_spans (0.21s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestCorruptData PKG=./pkg/kv/kvserver/protectedts/ptstorage TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/kv ajwerner
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCorruptData.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/protectedts/ptstorage.TestCorruptData [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3690958&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3690958&tab=artifacts#/) on master @ [46b7efa32c57b15f4ae6e0d75845f63360b62161](https://github.com/cockroachdb/cockroach/commits/46b7efa32c57b15f4ae6e0d75845f63360b62161):


```
=== RUN   TestCorruptData
--- FAIL: TestCorruptData (0.39s)
=== RUN   TestCorruptData/corrupt_spans
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCorruptData_corrupt_spans802016722
    test_log_scope.go:80: use -show-logs to present logs inline
    storage_test.go:574: 
        	Error Trace:	storage_test.go:574
        	Error:      	"[{ERROR %!s(int64=1636182178601228000) %!s(int64=19761) kv/kvserver/protectedts/ptstorage/storage.go %!s(int64=240) failed to parse row as record: failed to unmarshal spans for eebcb81f-0d1a-4ee1-9416-21d4f1cb021a: unexpected EOF - %!s(uint64=51) %!s(bool=false) DEV %!s(uint32=0) %!s(uint32=0) %!s(uint32=0)} {ERROR %!s(int64=1636182178598992000) %!s(int64=18580) kv/kvserver/protectedts/ptstorage/storage.go %!s(int64=240) failed to parse row as record: failed to unmarshal spans for eebcb81f-0d1a-4ee1-9416-21d4f1cb021a: unexpected EOF - %!s(uint64=50) %!s(bool=false) DEV %!s(uint32=0) %!s(uint32=0) %!s(uint32=0)}]" should have 1 item(s), but has 2
        	Test:       	TestCorruptData/corrupt_spans
    panic.go:613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCorruptData_corrupt_spans802016722
    --- FAIL: TestCorruptData/corrupt_spans (0.24s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCorruptData.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/protectedts/ptstorage/storage_test.go
+++ b/pkg/kv/kvserver/protectedts/ptstorage/storage_test.go
@@ -571,7 +571,7 @@ func TestCorruptData(t *testing.T) {
 		entries, err := log.FetchEntriesFromFiles(0, math.MaxInt64, 100, msg,
 			log.WithFlattenedSensitiveData)
 		require.NoError(t, err)
-		require.Len(t, entries, 1)
+		require.GreaterOrEqual(t, 1, len(entries), "entries: %v", entries)
 		for _, e := range entries {
 			require.Equal(t, severity.ERROR, e.Severity)
 		}
@@ -623,7 +623,7 @@ func TestCorruptData(t *testing.T) {
 		entries, err := log.FetchEntriesFromFiles(0, math.MaxInt64, 100, msg,
 			log.WithFlattenedSensitiveData)
 		require.NoError(t, err)
-		require.Len(t, entries, 1)
+		require.GreaterOrEqual(t, 1, len(entries), "entries: %v", entries)
 		for _, e := range entries {
 			require.Equal(t, severity.ERROR, e.Severity)
 		}

